doc/source/dev/contributor/continuous_integration.rst

```diff
@@ -119,3 +119,22 @@ It is not advised to use cibuildwheel to build scipy wheels on your own system
 as it will automatically install gfortran compilers and various other
 dependencies. Instead, one could use an isolated Docker container to build
 Linux wheels.
+
+Docker image builds
+===================
+
+Two Docker images are built in CI with GitHub Actions, and made available on
+Docker Hub under the ``scipy`` organisation:
+
+1. ``scipy/scipy-dev`` - contains all the dependencies needed for developing
+   SciPy. This image is only rebuilt when ``environment.yml`` in the root of
+   the repo is updated,
```

Comment:
I think we should add more warnings there. As we start to advertise that we do have images, some people might be tempted to used these on production systems. If we rarely build, we are basically having more risk to have a compromised image.

Reply:
Warnings don't make sense to me. We do rebuild it every so often (my guess is typical frequency is every 6 weeks on average), it is not meant for production systems and not advertised as such (why would anyone want scipy dev dependencies in a production system?), and a 6 weeks old image really isn't that risky anyway.

Reply:
Agree with Ralf - the images have dev dependencies for SciPy. I would not imagine folks wanting to use this for production systems. Plus a 6 week cadence is more than enough to pull security patches in any Ubuntu owned images

Reply:
As you wish.

This can really end up in production. If someone needs to build something custom into SciPy, they could see this image as a low effort into ensuring they have a proper setup to do that. I have seen similar setups (needed to rebuild a famous package for some reasons and it was a pain to find how to do this reliably) and far from all teams have good practices to rebuild images frequently, do proper isolation, networking, stages, etc. Being SciPy, what we ship is really seen as a proper way to do things and I can really see these images being seen as safe because they are coming from us (I have heard this argument so many times as to blindly trust something just on the basis that it's from a known authority). As for the 6 weeks, it's an eternity in security.

Again, I am not saying we should explain anything. Just saying that this is not for production and you should do your own due diligence.

Reply:
I know - but the way the Docker images are built is quite hard for folks to use directly into any prod system. At least they would need to do some heavy work to make them work 🤷🏽‍♀️ .

If anything for peace of mind we can add a lightweight disclaimer:

"Note these Docker images are intended for SciPy development *only*. These should not be used for any production-level applications (in which case SciPy cannot be held responsible as per  our license.....)"


or similar - WDYT?

Reply:
That would be better IMHO. Thanks Tania.

Reply:
Thanks for flagging this @tupui  - it is easy to forget that folks might find unintended uses for the tools we build at any time. And if anything adding a disclaimer like this will bring peace of mind to maintainers 🙏🏽